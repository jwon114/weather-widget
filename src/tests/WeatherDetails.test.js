import React from 'react';
import renderer from 'react-test-renderer';
import WeatherDetails from '../components/WeatherDetails';

test('renders WeatherDetails', () => {
  const tree = renderer
    .create(<WeatherDetails 
              icon={'01d'}
              temperature={20}
              precipitation={0}
              humidity={20}
              windSpeed={5}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});