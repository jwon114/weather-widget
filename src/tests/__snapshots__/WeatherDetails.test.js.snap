// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WeatherDetails 1`] = `
<div
  className="weatherDetails__main"
>
  <div
    className="weatherDetails__left_container"
  >
    <div
      className="weatherDetails__icon"
    >
      <img
        alt="weather forecast icon"
        src="http://openweathermap.org/img/wn/01d@2x.png"
      />
    </div>
    <div
      className="weatherDetails__temperature_container"
    >
      <span
        className="weatherDetails__temperature"
      >
        20
      </span>
      <div
        className="weatherDetails__unit_click active"
      >
        <span
          className="weatherDetails__unit"
          onClick={[Function]}
        >
          °C
        </span>
      </div>
      <span
        className="weatherDetails__vert_separator"
      >
        |
      </span>
      <div
        className="weatherDetails__unit_click "
      >
        <span
          className="weatherDetails__unit"
          onClick={[Function]}
        >
          °F
        </span>
      </div>
    </div>
  </div>
  <div
    className="weatherDetails__right_container"
  >
    <div
      className="weatherDetails__precipitation"
    >
      Precipitation: 
      0
      mm
    </div>
    <div
      className="weatherDetails__humidity"
    >
      Humidity: 
      20
      %
    </div>
    <div
      className="weatherDetails__wind"
    >
      Wind: 
      18
       km/h
    </div>
    <div
      className="weatherDetails__conditions"
    >
      <button
        className="weatherDetails__conditions_button"
      >
        Temperature
      </button>
      <button
        className="weatherDetails__conditions_button"
      >
        Precipitation
      </button>
      <button
        className="weatherDetails__conditions_button"
      >
        Wind
      </button>
    </div>
  </div>
</div>
`;
